### PR TITLE
Manage the Cypress tests entirely with Yarn

### PR DESCRIPTION
**Pull request by leofeyer**
<del>This PR adds a small runner script, which allows us to setup up everything and to use command line flags to control the execution; currently `yarn e2e` and `yarn e2e -i`, which will open the interactive Cypress GUI.</del>

### CODE REVIEW

**Comment by ausi** [comment]
> `yarn e2e` and `yarn e2e -i`

Will this work on your Windows setup @fritzmg as it uses a shell script?

**Comment by fritzmg** [comment]
`yarn e2e` won't work - but just running `./cypress/run.sh` will in Git Bash for example. But obviously won't work on CMD.

**Comment by leofeyer** [comment]
<del>`yarn e2e` is a shortcut for `yarn run e2e`. It should work everywhere. 🤔 </del>

I got the point now. How does Composer solve this?

**Comment by fritzmg** [comment]
> How does Composer solve this?

Composer creates proxy batch files for every binary registered in the `composer.json`. This is what `contao-console.bat` looks like:

```
@ECHO OFF
setlocal DISABLEDELAYEDEXPANSION
SET BIN_TARGET=%~dp0/contao-console
SET COMPOSER_RUNTIME_BIN_DIR=%~dp0
php "%BIN_TARGET%" %*
```

https://getcomposer.org/doc/articles/vendor-binaries.md#what-about-windows-and-bat-files-

**Comment by leofeyer** [comment]
As discussed in the Contao call, we cannot use a Bash runner.

File: package.json [hunk 1]
```diff
@@ -16,16 +16,11 @@
     },
     "private": true,
     "scripts": {
-        "dev-server": "encore dev-server",
-        "dev": "encore dev",
-        "watch": "encore dev --watch",
+        "e2e": "cypress/run.sh",
```
Comment (ausi) [inline package.json]:
```suggestion
        "e2e": "rm -rf \"cypress/webspace/vendor/contao/manager-bundle\" && composer up --working-dir=\"cypress/webspace\" && symfony server:start --port=8765 --dir=\"cypress/webspace\" --daemon && yarn cypress run --config-file \"cypress/cypress.config.js\" && symfony server:stop --dir=\"cypress/webspace\"",
```

something like this?

**Comment by leofeyer** [comment]
@fritzmg Can you please check if the current solution with the `cypress/run.js` file works on Windows?

**Comment by fritzmg** [comment]
It does work on the Git Bash - but still won't work on CMD:

```
  Backend
    1) "before all" hook for "Create a Website"


  0 passing (2s)
  1 failing

  1) Backend
       "before all" hook for "Create a Website":
     CypressError: `cy.exec('rm -rf "cypress/webspace/files/"')` failed because the command exited with a non-zero code.
Pass `{failOnNonZeroExit: false}` to ignore exit code failures.

Information about the failure:
Code: 1

Stderr:
'rm' is not recognized as an internal or external command,
operable program or batch file.

https://on.cypress.io/exec

Because this error occurred during a `before all` hook we are skipping the remaining tests in the current suite: `Backend`
      at <unknown> (https://localhost:8765/__cypress/runner/cypress_runner.js:146202:85)
      at tryCatcher (https://localhost:8765/__cypress/runner/cypress_runner.js:18744:23)
      at Promise._settlePromiseFromHandler (https://localhost:8765/__cypress/runner/cypress_runner.js:16679:31)
      at Promise._settlePromise (https://localhost:8765/__cypress/runner/cypress_runner.js:16736:18)
      at Promise._settlePromise0 (https://localhost:8765/__cypress/runner/cypress_runner.js:16781:10)
      at Promise._settlePromises (https://localhost:8765/__cypress/runner/cypress_runner.js:16861:18)
      at _drainQueueStep (https://localhost:8765/__cypress/runner/cypress_runner.js:13451:12)
      at _drainQueue (https://localhost:8765/__cypress/runner/cypress_runner.js:13444:9)
      at ../../node_modules/bluebird/js/release/async.js.Async._drainQueues (https://localhost:8765/__cypress/runner/cypress_runner.js:13460:5)
      at Async.drainQueues (https://localhost:8765/__cypress/runner/cypress_runner.js:13330:14)
  From Your Spec Code:
      at Context.eval (webpack:///./support/commands.js:13:7)
```

But I think that's fine (well unless we want to run this via CI/CD?)

**Comment by leofeyer** [comment]
But the current solution in the `5.2` branch (without any runner script) does work on Windows?

**Comment by fritzmg** [comment]
Actually no, I think I only tested in the Git Bash back then. Via CMD it also does not work in the current `5.2` branch.

**Comment by leofeyer** [comment]
5850214f28c95fd264b5a7c848dd7f44b61daf5f is the final version now. It does not use a runner script but it does require the Git shell on Windows. No way this will ever work with CMD. 🤷‍♂️ 

```
yarn install
yarn run e2e-tests
```

This is the same concept that we are using for the Composer scripts as well:

```
composer bin all install
composer run unit-tests
```

**Pull request merged**